core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -1360,10 +1363,11 @@ public function getFrontendUrl($strParams=null, $strForceLang=null)
 		}
 
 		$objRouter = System::getContainer()->get('router');
+		$referenceType = $this->domain && $objRouter->getContext()->getHost() !== $this->domain ? UrlGeneratorInterface::ABSOLUTE_URL : UrlGeneratorInterface::ABSOLUTE_PATH;
 
 		try
 		{
-			$strUrl = $objRouter->generate(PageRoute::PAGE_BASED_ROUTE_NAME, array(RouteObjectInterface::CONTENT_OBJECT => $this, 'parameters' => $strParams));
+			$strUrl = $objRouter->generate(PageRoute::PAGE_BASED_ROUTE_NAME, array(RouteObjectInterface::CONTENT_OBJECT => $this, 'parameters' => $strParams), $referenceType);
```

Comment:
Since `$strUrl` cannot be a network path anymore now, we can remove the ` && 0 !== strncmp($strUrl, '//', 2)` check in line 1385, can‘t we?